cdap-runtime-ext-remote-hadoop/src/main/java/io/cdap/cdap/runtime/spi/provisioner/remote/RemoteHadoopProvisioner.java

```diff
@@ -158,6 +160,19 @@ String selectEdgeNode(String hostConfigValue, String profileName, ProgramRunInfo
     return host;
   }
 
+  String selectCheckedEdgeNode(List<String> hosts, int initialIndex, EdgeNodeCheckType edgeNodeCheckType, int timeout)
+    throws NoLiveEdgeNodeException {
+    LOG.trace(edgeNodeCheckType + " check type is used");
+    switch (edgeNodeCheckType) {
+      case NONE:
+        return hosts.get(initialIndex % hosts.size());
+      case PING:
+        return new PingEdgeNodeCheck().selectPingableEdgeNode(hosts, initialIndex, timeout);
```

Comment:
Can `selectPingableEdgeNode` be a static method? If no, than probably we should create an instance of `PingEdgeNodeCheck` to avoid it creating each time.

Reply:
Done